### PR TITLE
add bz2 from CPython 3.10.5

### PR DESCRIPTION
**Pull request by verhovsky**
pretty much copy and pasted from `csv.rs`.

### CODE REVIEW

**Comment by verhovsky** [comment]
Not done, I get this error

```
error[E0277]: the trait bound `for<'r, 's> fn(&'r pyobjectrc::PyRef<BZ2Decompressor>, byteslike::PyBytesLike, function::OptionalArg<i32>, &'s vm::VirtualMachine) -> std::result::Result<pyobjectrc::PyObjectRef, pyobjectrc::PyRef<PyBaseException>> {BZ2Decompressor::decompress}: PyNativeFuncInternal<_, _, _>` is not satisfied
  --> vm/src/stdlib/bz2.rs:31:1
   |
31 | #[pyimpl]
   | ^^^^^^^^^ the trait `PyNativeFuncInternal<_, _, _>` is not implemented for `for<'r, 's> fn(&'r pyobjectrc::PyRef<BZ2Decompressor>, byteslike::PyBytesLike, function::OptionalArg<i32>, &'s vm::VirtualMachine) -> std::result::Result<pyobjectrc::PyObjectRef, pyobjectrc::PyRef<PyBaseException>> {BZ2Decompressor::decompress}`
   |
   = note: required because of the requirements on the impl of `IntoPyNativeFunc<(_, _, _)>` for `for<'r, 's> fn(&'r pyobjectrc::PyRef<BZ2Decompressor>, byteslike::PyBytesLike, function::OptionalArg<i32>, &'s vm::VirtualMachine) -> std::result::Result<pyobjectrc::PyObjectRef, pyobjectrc::PyRef<PyBaseException>> {BZ2Decompressor::decompress}`
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error
```

can you tell me what I did wrong?

File: vm/src/stdlib/bz2.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    })
+}
+
+pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
+    let ctx = &vm.ctx;
+    BZ2Decompressor::make_class(ctx);
+    BZ2Compressor::make_class(ctx);
+
+    py_module!(vm, "_bz2", {
+        "BZ2Decompressor" => named_function!(ctx, _bz2, BZ2Decompressor),
```
Comment (coolreader18) [inline vm/src/stdlib/bz2.rs:210]: I know this is a WIP, but just a note - it looks like CPython exports `_bz2.BZ2[De]compressor` as the classes, rather as separate functions. You can move the functionality from `_bz2_BZ2Decompressor` to an associated function `BZ2Decompressor::tp_new`, with a `#[pyslot]` attribute on it, and same with `BZ2Compressor`. Then you can export directly as `"BZ2Decompressor" => BZ2Decompressor::make_class(ctx)`

File: stdlib/src/lib.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@ mod random;
 mod statistics;
 // TODO: maybe make this an extension module, if we ever get those
 // mod re;
+#[cfg(not(any(target_arch = "wasm32")))]
```
Comment (fanninpm) [inline stdlib/src/lib.rs:28]:
```suggestion
#[cfg(not(target_arch = "wasm32"))]
```

**Review by youknowone — COMMENTED** [review]
not done but harmless to merge

**Pull request merged**